### ISSUE TITLE
error: call to undeclared function 'PyDataType_ELSIZE'; ISO C99 and later do not support implicit function declarations [-Werror,-Wimplicit-function-declaration]

### ISSUE THREAD

**Issue by yurivict** [dofuuz/python-soxr#31]
Build fails:
```
src/soxr/cysoxr.c:18786:13: error: call to undeclared function 'PyDataType_ELSIZE'; ISO C99 and later do not support implicit function declarations [-Werror,-Wimplicit-function-declaration]
  __pyx_r = PyDataType_ELSIZE(__pyx_v_self);
            ^
src/soxr/cysoxr.c:18820:13: error: call to undeclared function 'PyDataType_ALIGNMENT'; ISO C99 and later do not support implicit function declarations [-Werror,-Wimplicit-function-declaration]
  __pyx_r = PyDataType_ALIGNMENT(__pyx_v_self);
            ^
src/soxr/cysoxr.c:18858:15: error: call to undeclared function 'PyDataType_FIELDS'; ISO C99 and later do not support implicit function declarations [-Werror,-Wimplicit-function-declaration]
  __pyx_t_1 = PyDataType_FIELDS(__pyx_v_self);
              ^
src/soxr/cysoxr.c:18858:13: error: incompatible integer to pointer conversion assigning to 'PyObject *' (aka 'struct _object *') from 'int' [-Wint-conversion]
  __pyx_t_1 = PyDataType_FIELDS(__pyx_v_self);
            ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Version: 0.4.0
Python-3.11
clang-18
FreeBSD 14.1

**Comment by dofuuz** [comment]
Version of NumPy and Cython?

**Comment by dofuuz** [comment]
May be related:
https://github.com/cython/cython/issues/6257

**Comment by yurivict** [comment]
py310-numpy-1.26.4
py311-cython3-3.0.10

**Comment by dofuuz** [comment]
python-soxr v0.4.0 uses NumPy 2 for building.
If you need to use v0.4.0 with NumPy 1.x, get git source, modify pyproject.toml and build it.

But v0.3.7 and v0.4.0 has no difference except NumPy 2 compatibility. So I suggest to stay v0.3.7 if you need to use NumPy 1.x.

**Comment by dofuuz** [comment]
Fixed with v0.5.0a1
